src/core/api/srp_server_api.cpp

```diff
@@ -147,12 +147,16 @@ uint16_t otSrpServerServiceGetPriority(const otSrpServerService *aService)
     return static_cast<const Srp::Server::Service *>(aService)->GetPriority();
 }
 
-otError otSrpServerServiceGetNextTxtEntry(const otSrpServerService *aService,
-                                          otDnsTxtIterator *        aIterator,
-                                          otDnsTxtEntry *           aTxtEntry)
+const uint8_t *otSrpServerServiceGetTxtData(const otSrpServerService *aService, uint16_t *aDataLength)
 {
-    return static_cast<const Srp::Server::Service *>(aService)->GetNextTxtEntry(
-        *aIterator, static_cast<Dns::TxtEntry &>(*aTxtEntry));
+    const Srp::Server::Service &service = *static_cast<const Srp::Server::Service *>(aService);
+
+    if (aDataLength != nullptr)
+    {
+        *aDataLength = service.GetTxtDataLength();
+    }
```

Comment:
nit: I think there are no cases that the caller should not get the TXT length, so may just crash if `NULL` is provided. We can document this non-NULL requirement for the `ot` API.
```suggestion
    *aDataLength = service.GetTxtDataLength();
```

Reply:
SGTM.